rpcclient/infrastructure.go

```diff
@@ -905,7 +942,7 @@ out:
 cleanup:
 	for {
 		select {
-		case jReq := <-c.highPriorityPostQueue:
+		case jReq := <-c.nextPostRequest():
```

Comment:
Same here, this breaks assumptions and will block forever on shutdown.

Reply:
here is a bit different as we wanna drain the channel, so changed to read the channels directly.